enhancements/multi-arch/heterogeneous-architecture.md

```diff
@@ -0,0 +1,373 @@
+upgrade.
+
+Cincinnati needs to [support][OCPPLAN-7643] parsing manifestlist images to be able to upgrade to a heterogeneous architecture payload. There will be a
+new architecture option added to it called "multi".
+
+### Imagestreams
+
+Imagestreams need to better support manifestlist images. As of today, the imagestreams support only import of the image matching the control plane
+architecture. This is due to the fact the internal image registry does not support manifestlisted images and that is not in the scope of this
+implementation. Hence imagestream imports would have to work in such a way that they would not break pullthrough which is used in our CI tooling.
+At that same time it would also have to support manifestlist images in order to work on multi architecture nodes.
+
+In order to support this, the proposal is that the imagestreams would be manifest listed. The API would provide an option to "preserve" the
+manifestlist rather than importing a single image. But in order to support pullthrough from the internal image registry, there will be a new child
+manifest introduced in the imagestream which will contain the image matching that of the control plane architecture. For imports into the image
+registry, a manifestlist containing a single image matching that of the control plane architecture will be imported (if the option to preserve
+manifestlist is chosen. if not it would be a single manifest image). This ensures that the image registry behavior remains as is and current CI
+tooling is not affected. At the same time, for non control plane architecture workers, while the image  registry images would not work, they can
+be overriden with the manifest listed imagestream. Eventually the oc tooling would also be upgraded to point to the manifest listed image from
+source for non control plane architecture workers.
+
+### OLM
+
+OLM would need to make sure that it deploys operators only to nodes whose architecture matches the architectures on which the operator is supported.
+This would mean that the OLM would have to be told which architectures an operator supports by the Console operator. Today, OLM has a subscription
+object which is used to tell it how to deploy an operator. It has support for the nodeselector field but this would not be sufficient to specify
+filtering on multiple architectures. To this end, it would have to support the podAffinity field which uses the nodeselectorterms to filter on
+multiple values.
+
+Instead of having OLM dictate the operator deployment on specific nodes, the individual operators themselves would specify which architectures
+they support through a nodeselector object which OLM will then provide the ability for the user to override if necessary. This information should
+also be added to the operands and  potentially be made configurable by the user requesting the operand.The operator-sdk framework could also be
```

Comment:
Couple minor nits.

Extra space after `added to the operands and`
Space needed before `The operator-sdk`